projects/sicx/index.js

```diff
@@ -2,27 +2,56 @@ const axios = require("axios")
 
 const icxApiEndpoint = 'https://ctz.solidwallet.io/api/v3'
 const stakingContract = "cx43e2eec79eb76293c298f2b17aec06097be606e0"
+const sIcxTokenContract = "cx2609b924e33ef00b648a409245c7ea394c467824"
+const bandOracleContract = 'cx087b4164a87fdfb7b714f3bafe9dfb050fd6b132'
 
-async function getBalance(address) {
+async function icxCall(address, method, params) {
     let response = await axios.post(icxApiEndpoint, {
         jsonrpc: '2.0',
-        method: 'icx_getBalance',
+        method: 'icx_call',
         id: 1234,
         params: {
-            address,
+            to: address,
+            dataType: 'call',
+            data: {
+                method: method,
+                params: params
+            }
         }
     })
     return response.data.result
 }
 
+async function getIcxUsdPrice() {
+    let response = await icxCall(bandOracleContract, 'get_ref_data', { '_symbol': 'ICX' })
+    let icxUsdPrice = sicxIcxPrice = parseInt(response.rate, 16) / 10 ** 9;
+    return icxUsdPrice
+}
+
 async function getTotalClaimableIcx() {
-    let response = await getBalance(stakingContract)
+    let response = await icxCall(stakingContract, 'totalClaimableIcx')
+    return parseInt(response, 16) / 10 ** 18
+}
+
+async function getTotalSupply() {
+    let response = await icxCall(sIcxTokenContract, 'totalSupply')
+    return parseInt(response, 16) / 10 ** 18
+}
+
+async function getTodayRate() {
+    let response = await icxCall(stakingContract, 'getTodayRate')
     return parseInt(response, 16) / 10 ** 18
 }
 
 async function fetch() {
-    let totalClaimableIcx = await getTotalClaimableIcx()
-    return { 'icon': totalClaimableIcx }
+
+    let icxPrice = await getIcxUsdPrice();
+    let totalClaimableIcx = await getTotalClaimableIcx();
+    let totalSupply = await getTotalSupply();
+    let todayRate = await getTodayRate();
+    // tvl = ICX/USD Price * (totalClaimableIcx + totalSupply from sICX token SCORE * getTodayRate)
+    const tvl = icxPrice * (totalClaimableIcx + totalSupply * todayRate);
```

Comment:
I think this calculation is bit off, we are returning amount of icon tokens here, and we later fetch the price from coingecko, so no need to fetch the price

Reply:
Hi @g1nt0ki, we the calculation should be `totalClaimableIcx + totalSupply * todayRate`. Please help to update it. Thank you

Reply:
@shallcdn I am a bit confused, wouldnt `totalSupply * todayRate` include the claimable icx as well? I see no option for end user who converted ICX -> sICX to claim anything

Reply:
> @g1nt0ki Let me explain a bit. As you see in the code we have 2 contract:
> 
> **sICX token contract** = cx2609b924e33ef00b648a409245c7ea394c467824 **Staking Management contract** = cx43e2eec79eb76293c298f2b17aec06097be606e0
> 
> `totalSupply` got from **sICX token contract** it means total sICX already minted `totalClaimableIcx` got from **Staking Management contract** it means total ICX already staked in this contract,
> 
> Further, the price of sICX is higher than ICX, so we have `todayRate` to get the rate between ICX and sICX and then multiply `totalSupply`
> 
> As we know, sICX is staked ICX, so to the tvl of sICX will be calculated: `ICX/USD Price * (totalClaimableIcx + totalSupply from sICX token SCORE * getTodayRate`

I understand, but my point is, I am seeing totalClaimableIcx (which is staked ICX?) this quantity is way below sICX total supply, is the rest of ICX is in some other contract for staking?

if I do `totalSupply * getTodayRate`, wont it give me all the ICX staked + rewards accumulated ? if so, shouldnt we leave out `totalClaimableIcx` from tvl calculation?